catalyst_voices/lib/pages/registration/recover/seed_phrase/seed_phrase_input_panel.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:catalyst_voices/pages/registration/recover/bloc_recover_builder.dart';
+import 'package:catalyst_voices/pages/registration/widgets/registration_stage_message.dart';
+import 'package:catalyst_voices/pages/registration/widgets/registration_stage_navigation.dart';
+import 'package:catalyst_voices/pages/registration/widgets/seed_phrase_actions.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_blocs/catalyst_voices_blocs.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:flutter/material.dart';
+
+class SeedPhraseInputPanel extends StatefulWidget {
+  const SeedPhraseInputPanel({super.key});
+
+  @override
+  State<SeedPhraseInputPanel> createState() => _SeedPhraseInputPanelState();
+}
+
+class _SeedPhraseInputPanelState extends State<SeedPhraseInputPanel> {
+  late final SeedPhraseFieldController _controller;
+  final _focusNode = FocusNode();
+
+  @override
+  void initState() {
+    super.initState();
+    final recoverState = RegistrationCubit.of(context).state.recoverStateData;
+    final words = recoverState.userSeedPhraseWords;
+
+    _controller = SeedPhraseFieldController(words)
+      ..addListener(_onWordsChanged);
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        const SizedBox(height: 24),
+        RegistrationStageMessage(
+          title: Text(context.l10n.recoverySeedPhraseInputTitle),
+          subtitle: Text(context.l10n.recoverySeedPhraseInputSubtitle),
+          spacing: 12,
+        ),
+        const SizedBox(height: 12),
+        Expanded(
+          child: _BlocSeedPhraseField(
+            controller: _controller,
+            focusNode: _focusNode,
+          ),
+        ),
+        const SizedBox(height: 12),
+        SeedPhraseActions(
+          onUploadKeyTap: _uploadSeedPhrase,
+          onResetTap: _resetControllerWords,
+        ),
+        const SizedBox(height: 12),
+        const _BlocNavigation(),
+      ],
+    );
+  }
+
+  Future<void> _uploadSeedPhrase() async {
+    // TODO(damian-molinski): Import implementation for KeychainCreation
+  }
+
+  void _resetControllerWords() {
+    _controller.clear();
+    _focusNode.requestFocus();
+  }
+
+  void _onWordsChanged() {
+    final words = [..._controller.value];
```

Comment:
If your intention was to create a copy of a list then List.of() seems a better alternative then using a spread operator